samples/dapr/AppHost/Program.cs

```diff
@@ -2,11 +2,17 @@
 
 builder.AddDapr();
 
+var stateStore = builder.AddDaprStateStore("statestore");
+var pubSub = builder.AddDaprPubSub("pubsub");
+
 builder.AddProject<Projects.DaprServiceA>("servicea")
-       .WithDaprSidecar("service-a");
+       .WithDaprSidecar("service-a")
```

Comment:
It looks like you've got an overloaded version of the `WithReference` extension method. Given you are doing that I'm wondering whether we need the `WithDaprSidecar` call? Couldn't your custom implementation of `WithReference check to see that the annotation is present (and could the name be derived?).

Reply:
Do you mean that, when referencing components, that the need for the sidecar can be inferred?

Reply:
Yeah that is what I was thinking. Just removing a bit of (potentially) unnecessary ceremony?